### PR TITLE
Implemented EventOnChampionMove in ApiEventManager.cs and Champion.cs

### PR DESCRIPTION
**Pull request by abosma**
Implemented EventOnChampionMove in ApiEventManager called by Champion.UpdateMoveOrder. This allows for channels like Recall to be cancelled on user movement.

### CODE REVIEW

File: GameServerLib/GameObjects/AttackableUnits/AI/Champion.cs [hunk 1]
```diff
@@ -167,6 +167,8 @@ public bool CanMove()
         public void UpdateMoveOrder(MoveOrder order)
         {
             MoveOrder = order;
+
+            ApiEventManager.OnChampionDamageTaken.Publish(this);
```
Comment (FurkanS1821) [inline GameServerLib/GameObjects/AttackableUnits/AI/Champion.cs]: Um, you move and you trigger damage taken event? Okay.

**Review by FurkanS1821 — DISMISSED** [review]
🤔

File: GameServerLib/GameObjects/AttackableUnits/AI/Champion.cs (hunk 1 above)
Comment (abosma) [reply GameServerLib/GameObjects/AttackableUnits/AI/Champion.cs]: Resolved and added Unit and Champion crowd control event handlers

**Review by danil179 — APPROVED** [review]
Approved.

**Pull request merged**